rama-cli/src/cmd/tls/mod.rs

```diff
@@ -0,0 +1,152 @@
+use clap::Args;
+use rama::{
+    Context, Layer, Service,
+    error::BoxError,
+    net::{
+        address::Authority,
+        client::{ConnectorService, EstablishedClientConnection},
+        tls::{
+            DataEncoding,
+            client::{ClientConfig, NegotiatedTlsParameters, ServerVerifyMode},
+        },
+    },
+    tcp::client::{Request, service::TcpConnector},
+    tls::{
+        rustls::client::{TlsConnectorData, TlsConnectorLayer},
+        std::dep::boring::x509::X509,
+    },
+};
+use tokio::net::TcpStream;
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::{EnvFilter, fmt, layer::SubscriberExt, util::SubscriberInitExt};
+
+#[derive(Args, Debug, Clone)]
+/// rama tls support
+pub struct CliCommandTls {
+    /// The address to connect to
+    /// e.g. "example.com" or "example.com:8443"
+    /// if no port is provided, the default port 443 will be used
+    address: String,
+
+    #[arg(long, short = 'k')]
+    /// Wether to skip certificate verification
+    insecure: bool,
+}
+
+/// Run the tls command
+pub async fn run(cfg: CliCommandTls) -> Result<(), BoxError> {
+    tracing_subscriber::registry()
+        .with(fmt::layer())
+        .with(
+            EnvFilter::builder()
+                .with_default_directive(LevelFilter::INFO.into())
+                .from_env_lossy(),
+        )
+        .init();
+
+    let authority = match Authority::try_from(cfg.address.clone()) {
+        Ok(authority) => authority,
+        // if missing port, we can try to add the default port
+        Err(err) if err.to_string().contains("missing port") => {
+            let authority = format!("{}:443", cfg.address);
+            Authority::try_from(authority)?
+        }
+        Err(err) => return Err(err.into()),
+    };
+
+    tracing::info!("Connecting to: {}", authority);
+
+    let tls_client_data = TlsConnectorData::try_from(ClientConfig {
+        store_server_certificate_chain: true,
+        server_verify_mode: if cfg.insecure {
+            Some(ServerVerifyMode::Disable)
+        } else {
+            Some(ServerVerifyMode::Auto)
+        },
+        ..Default::default()
+    })
+    .expect("create tls connector data for client");
+
+    let tcp_connector = TcpConnector::new();
+    let loggin_service = LoggingLayer.layer(tcp_connector);
+
+    let tls_connector = TlsConnectorLayer::secure()
+        .with_connector_data(tls_client_data)
+        .layer(loggin_service);
+
+    let EstablishedClientConnection {
+        ctx,
+        req: _,
```

Comment:
you can just do:

```
EstablishedClientConnection { ctx, .. }
```

Forgot if it's 2 dots or 3 dots, but either way that does exactly what you want without having to list all the properties you do not care about.